blobstore/src/main/java/org/jclouds/blobstore/domain/internal/StorageMetadataImpl.java

```diff
@@ -60,6 +64,23 @@ public StorageMetadataImpl(StorageType type, @Nullable String id, @Nullable Stri
       this.type = checkNotNull(type, "type");
       this.size = size;
       this.tier = tier;
+      this.versionId = null;
+      this.isLatest = null;
+   }
+
+   public StorageMetadataImpl(StorageType type, @Nullable String id, @Nullable String name,
+         @Nullable Location location, @Nullable URI uri, @Nullable String eTag,
+         @Nullable Date creationDate, @Nullable Date lastModified,
+         Map<String, String> userMetadata, @Nullable Long size, Tier tier, @Nullable String versionId) {
+      super(id, name, location, uri, userMetadata);
```

Comment:
Please make the constructor with all parameters the only one that sets fields and the existing one with fewer parameters can call `this(...)` with a default value for `versionId`.